recipes/jasper/all/conanfile.py

```diff
@@ -38,27 +42,30 @@ def _build_subfolder(self):
 
     def export_sources(self):
         self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.cppstd
-        del self.settings.compiler.libcxx
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
 
     def requirements(self):
         if self.options.with_libjpeg == "libjpeg-turbo":
             self.requires("libjpeg-turbo/2.1.2")
         elif self.options.with_libjpeg == "libjpeg":
-            self.requires("libjpeg/9d")
+            self.requires("libjpeg/9e")
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) >= "16":
+            raise ConanInvalidConfiguration(f"{self.name} Current can not build in CCI due to windows SDK version. See https://github.com/conan-io/conan-center-index/pull/13285 for the solution hopefully")
```

Comment:
I don't think it's acceptable, it has too many consequences in downstream recipes.

Reply:
Agree. The problem is just in the CI configuration. If a user has the latest SDK installed on a local machine, which is highly possible, the issue will not raise locally.

Reply:
Thanks for your input.

This is not new and we do it in tons of recipes.
You can always host your own copies of the recipes while you wait for this issue to be resolved.

Happy contributing!